components/grid/selection/rows.md

```diff
@@ -0,0 +1,276 @@
+* When you delete a selected item in the Grid, it will automatically delete from the `SelectedItems` collection. If you are using one-way binding for the `SelectedItems` collection and the [`SelectedItemsChanged` event](#selecteditemschanged), when you delete a selected item, the event fires. When you delete all selected items, the `SelectedItemsChanged` event fires with an empty collection.
+* When you create an item in the Grid, and you want to select it at the same time, use the Grid [editing events]({%slug components/grid/editing/overview%}).
+
+### Selected Rows Equals Comparison
+
+The `SelectedItems` collection is compared against the Grid `Data` collection in order to determine which rows will be highlighted. The default behavior of the framework is to compare objects by their reference.
+
+When the `SelectedItems` are obtained from a different data source to the Grid (e.g., from a separate service method and not from the view-model), the references may not match and so there will be no highlighted items. In such cases, you have to [override the `Equals` method of the underlying model class]({%slug grid-state%}#equals-comparison) so that it matches them, for example, by a unique identifier rather than by reference so that two objects can be equal regardless of their origin, but according to their contents. When you are overriding the `Equals` method, it is also recommended to override the [`GetHashCode`](https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode) method as well.
+
+## SelectedItemsChanged
+
+You can respond to the user action of selecting a new row through the `SelectedItemsChanged` event. The `SelectedItemsChanged` event receives a collection of the Grid data model. It may have no items in it. It may have only one member (the last selected item) when the `SelectionMode` is `Single`.
+
+>caption Single selection with one-way binding for SelectedItems and using the SelectedItemsChanged event
+
+````CSHTML
+<TelerikGrid Data=@GridData
+             SelectionMode="@GridSelectionMode.Single"
+             SelectedItemsChanged="@((IEnumerable<Employee> employeeList) => OnSelect(employeeList))"
+             SelectedItems="@SelectedEmployees"
+             Pageable="true"
+             Height="400px">
+    <GridColumns>
+        <GridCheckboxColumn />
+        <GridColumn Field=@nameof(Employee.Name) />
+        <GridColumn Field=@nameof(Employee.Team) Title="Team" />
+    </GridColumns>
+</TelerikGrid>
+
+@if (SelectedEmployees != null)
+{
+    <ul>
+        @foreach (Employee employee in SelectedEmployees)
+        {
+            <li>
+                @employee.Name
+            </li>
+        }
+    </ul>
+}
+
+@code {
+    private List<Employee> GridData { get; set; }
+    private IEnumerable<Employee> SelectedEmployees { get; set; } = Enumerable.Empty<Employee>();
+    private Employee SelectedEmployee { get; set; }
```

Comment:
Objects should be nullable or instantiated immediately. If they are nullable, you may need to adjust other code in the example. The ultimate goal is to not have any compiler warnings.

```suggestion
    private Employee? SelectedEmployee { get; set; }
```